src/components/diagrams/singleLineDiagram/PositionDiagram.js

```diff
@@ -0,0 +1,441 @@
+        '& .sld-disconnector:not(.sld-fictitious), :not(.sld-breaker):not(.sld-disconnector):not(.sld-load-break-switch).sld-disconnected, .sld-feeder-disconnected, .sld-feeder-disconnected-connected':
+            {
+                stroke: theme.palette.text.primary,
+            },
+        '& .arrow': {
+            fill: theme.palette.text.primary,
+            pointerEvents: 'none',
+        },
+        '& .sld-in .sld-label': {
+            display: 'none',
+        },
+        '& .sld-out .sld-label': {
+            display: 'none',
+        },
+        '& .sld-flash, .sld-lock': {
+            stroke: 'none',
+            fill: theme.palette.text.primary,
+        },
+        '& .sld-arrow-in': {
+            display: 'none',
+        },
+        '& .sld-arrow-out': {
+            display: 'none',
+        },
+        overflow: 'hidden',
+    },
+    close: {
+        padding: 0,
+        borderRight: theme.spacing(1),
+    },
+    header: {
+        padding: 5,
+        display: 'flex',
+        flexDirection: 'row',
+        wordBreak: 'break-all',
+        backgroundColor: theme.palette.background.default,
+    },
+    paperBorders: {
+        borderLeft: '1px solid ' + theme.palette.action.disabled,
+        borderBottom: '1px solid ' + theme.palette.action.disabledBackground,
+        borderRight: '1px solid ' + theme.palette.action.hover,
+    },
+}));
+
+const noSvg = { svg: null, metadata: null, error: null, svgUrl: null };
+
+let initialWidth, initialHeight;
+// To allow controls that are in the corners of the map to not be hidden in normal mode
+// (but they are still hidden in fullscreen mode)
+const mapRightOffset = 120;
+const mapBottomOffset = 80;
+const borders = 2; // we use content-size: border-box so this needs to be included..
+// Compute the paper and svg sizes. Returns undefined if the preferred sizes are undefined.
+const computePaperAndSvgSizesIfReady = (
+    fullScreen,
+    svgType,
+    totalWidth,
+    totalHeight,
+    svgPreferredWidth,
+    svgPreferredHeight,
+    headerPreferredHeight
+) => {
+    if (
+        typeof svgPreferredWidth != 'undefined' &&
+        typeof headerPreferredHeight != 'undefined'
+    ) {
+        let paperWidth, paperHeight, svgWidth, svgHeight;
+
+        let maxWidth, maxHeight;
+        if (svgType === VOLTAGE_LEVEL_TYPE) {
+            maxWidth = maxWidthVoltageLevel;
+            maxHeight = maxHeightVoltageLevel;
+        }
+        svgWidth = Math.min(
+            svgPreferredWidth,
+            totalWidth - mapRightOffset,
+            maxWidth
+        );
+        svgHeight = Math.min(
+            svgPreferredHeight,
+            totalHeight - mapBottomOffset - headerPreferredHeight,
+            maxHeight
+        );
+        paperWidth = svgWidth + borders;
+        paperHeight = svgHeight + headerPreferredHeight + borders;
+
+        return { paperWidth, paperHeight, svgWidth, svgHeight };
+    }
+};
+
+const PositionDiagram = forwardRef((props, ref) => {
```

Comment:
There is so many code duplicated from SingleLineDiagram component. Have you tried to share a maximum of code? Especially the most tricky part regarding size calculation. Duplicating this code just increase the technical debt of GridStudy.